src/NUnitFramework/tests/Attributes/RangeAttributeTests.cs

```diff
@@ -22,478 +22,489 @@
         }
 
-        private void MethodWithMultipleFloatRange([Range(1.0f, 3.0f, 1.0f)] [Range(10.0f, 12.0f, 1.0f)] float x) { }
-
-        #endregion
-
-        #region Conversions
-
         [Test]
-        public void CanConvertIntRangeToShort()
+        public static void MinValueRange_Single()
         {
-            CheckValues(nameof(MethodWithIntRangeAndShortArgument), (short)1, (short)2, (short)3);
+            Assert.That(
+                GetData(new RangeAttribute(float.MinValue + SingleExtremaEpsilon * 2, float.MinValue, -SingleExtremaEpsilon), typeof(float)),
+                Is.EqualTo(new[] { float.MinValue + SingleExtremaEpsilon * 2, float.MinValue + SingleExtremaEpsilon, float.MinValue }));
         }
 
-        private void MethodWithIntRangeAndShortArgument([Range(1, 3)] short x) { }
-
         [Test]
-        public void CanConvertIntRangeToByte() 
+        public static void MinValueRangeLosingPrecision_Single()
         {
-            CheckValues(nameof(MethodWithIntRangeAndByteArgument), (byte)1, (byte)2, (byte)3);
+            Assert.That(() => GetData(new RangeAttribute(float.MinValue + SingleExtremaEpsilon, float.MinValue, -SingleExtremaEpsilon / 2), typeof(float)),
+                Throws.InstanceOf<ArithmeticException>());
         }
-        
-        private void MethodWithIntRangeAndByteArgument([Range(1, 3)] byte x) { }
 
         [Test]
-        public void CanConvertIntRangeToSByte() 
+        public static void MinValueRange_Double()
         {
-            CheckValues(nameof(MethodWithIntRangeAndSByteArgument), (sbyte)1, (sbyte)2, (sbyte)3);
+            Assert.That(
+                GetData(new RangeAttribute(double.MinValue + DoubleExtremaEpsilon * 2, double.MinValue, -DoubleExtremaEpsilon), typeof(double)),
+                Is.EqualTo(new[] { double.MinValue + DoubleExtremaEpsilon * 2, double.MinValue + DoubleExtremaEpsilon, double.MinValue }));
         }
-        
-        private void MethodWithIntRangeAndSByteArgument([Range(1, 3)] sbyte x) { }
 
         [Test]
-        public void CanConvertIntRangeToDecimal()
+        public static void MinValueRangeLosingPrecision_Double()
         {
-            CheckValues(nameof(MethodWithIntRangeAndDecimalArgument), 1M, 2M, 3M);
+            Assert.That(() => GetData(new RangeAttribute(double.MinValue + DoubleExtremaEpsilon, double.MinValue, -DoubleExtremaEpsilon / 2), typeof(double)),
+                Throws.InstanceOf<ArithmeticException>());
         }
-        
-        private void MethodWithIntRangeAndDecimalArgument([Range(1, 3)] decimal x) { }
 
         [Test]
-        public void CanConvertDoubleRangeToDecimal() 
+        public static void MinValueRange_Decimal()
         {
-            CheckValues(nameof(MethodWithDoubleRangeAndDecimalArgument), 1.0M, 1.1M, 1.2M);
+            const double step = (double)((-(decimal)LargestDoubleConvertibleToDecimal - -(decimal)NextLargestDoubleConvertibleToLowerDecimal) / 2);
```

Comment:
If we extracted `(decimal)NextLargestDoubleConvertibleToLowerDecimal` into a variable (with a shorter name) then perhaps the test would be easier to read? (also I actually had to debug the tests to have an idea of the value of `step` 😄 .

Reply:
I don't have the slightest clue what the value of `step` is. 😳